openedx/features/content_tagging/rules.py

```diff
@@ -35,27 +36,108 @@ def is_taxonomy_admin(user: User, taxonomy: oel_tagging.Taxonomy = None) -> bool
     return False
 
 
+def is_global_taxonomy_admin(user: User) -> bool:
+    """
+    Returns True if the given user is a Global Taxonomy Admin.
+
+    Global Taxonomy Admins include global staff and superusers.
+    """
+    if oel_tagging.is_taxonomy_admin(user):
+        return True
+
+
+def is_global_taxonomy_admin_or_course_creator(user: User) -> bool:
+    """
+    Returns True if the given user is a Global Taxonomy Admin or a course creator who can create courses for any org.
+    """
+    return is_global_taxonomy_admin(user) or is_content_creator(user, None)
+
+
+def is_taxonomy_admin(user: User, taxonomy: oel_tagging.Taxonomy) -> bool:
+    """
+    Returns True if the given user is a Taxonomy Admin for the given content taxonomy.
+
+    Check if the user is a Global Taxonomy Admin or a course creator who can create courses for any org.
+    Otherwise, a taxonomy must be provided to determine if the user is a org-level course creator for one of the
+    taxonomy's org owners.
+    """
+    if is_global_taxonomy_admin(user):
+        return True
+
+    if is_global_taxonomy(taxonomy):
+        return False  # Only global admins can edit global taxonomies
+
+    taxonomy_orgs = TaxonomyOrg.get_admin_organizations(
+        taxonomy=taxonomy,
+        rel_type=TaxonomyOrg.RelType.OWNER,
+    )
+    for org in taxonomy_orgs:
+        if is_content_creator(user, org.short_name):
+            return True
+    return False
+
+
+def is_global_taxonomy(taxonomy: oel_tagging.Taxonomy) -> bool:
+    """
+    Returns True if the given taxonomy is a global taxonomy.
+    """
+    return TaxonomyOrg.objects.filter(
+        taxonomy=taxonomy,
+        rel_type=TaxonomyOrg.RelType.OWNER,
+        org=None,
+    ).exists()
+
+
+def get_user_taxonomy_orgs(user: User) -> List[Organization]:
+    """
+    Return the list of organizations for which the given user is a taxonomy admin.
+    """
+    orgs = Organization.objects.all()
+
+    if is_global_taxonomy_admin(user):
+        return list(orgs)
+
+    return [org for org in orgs if is_content_creator(user, org.short_name)]
+
+
 @rules.predicate
 def can_view_taxonomy(user: User, taxonomy: oel_tagging.Taxonomy = None) -> bool:
     """
-    Anyone can view an enabled taxonomy,
-    but only taxonomy admins can view a disabled taxonomy.
+    Everyone can potentially view a taxonomy (taxonomy=None). The object permission must be checked
+    to determine if the user can view a specific taxonomy.
+    Only taxonomy admins can view a disabled taxonomy. #ToDo: Check if this is correct.
     """
-    if taxonomy:
-        taxonomy = taxonomy.cast()
-    return (taxonomy and taxonomy.enabled) or is_taxonomy_admin(user, taxonomy)
+    if not taxonomy:
+        return True
+
+    taxonomy = taxonomy.cast()
+
+    return is_taxonomy_admin(user, taxonomy) or taxonomy.enabled
```

Comment:
Performance-wise, it's cheaper to check `taxonomy.enabled`, so could you check that first?

```suggestion
    return taxonomy.enabled or is_taxonomy_admin(user, taxonomy)
```

Reply:
Done: 1541d0daee93dadca3168780122cf818f7e83353
I will pay more attention to that!